Gormsson/Source/GattCharacteristic/TemperatureMeasurement.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  TemperatureMeasurement.swift
+//  Gormsson
+//
+//  Created by Loïc GRIFFIE on 16/04/2019.
+//  Copyright © 2019 Loïc GRIFFIE. All rights reserved.
+//
+
+import CoreBluetooth
+
+internal final class TemperatureMeasurement: CharacteristicProtocol {
+    public var uuid: CBUUID {
+        return CBUUID(string: "2A1C")
+    }
+
+    public var service: GattService {
+        return .healthThermometer
+    }
+
+    public var format: DataInitializable.Type {
+        return TemperatureMeasurementType.self
+    }
+}
+
+public enum TemperatureMeasurementTypeEnum: UInt8, CustomStringConvertible {
+    case armpit = 1
+    case body
+    case ear
+    case finger
+    case gastroIntestinalTract
+    case mouth
+    case rectum
+    case toe
+    case tympanum
+
+    case unknown
+
+    public var description: String {
+        switch self {
+        case .armpit:
+            return "Armpit"
+        case .body:
+            return "Body (general)"
+        case .ear:
+            return "Ear (usually ear lobe)"
+        case .finger:
+            return "Finger"
+        case .gastroIntestinalTract:
+            return "Gastro-intestinal Tract"
+        case .mouth:
+            return "Mouth"
+        case .rectum:
+            return "Rectum"
+        case .toe:
+            return "Toe"
+        case .tympanum:
+            return "Tympanum (ear drum)"
+        default:
+            return "Reserved for future use"
+        }
+    }
+}
+
+public enum TemperatureMeasurementUnitEnum: UInt8, CustomStringConvertible {
+    case celsius = 0
+    case fahrenheit
+
+    case unknown
+
+    public var description: String {
+        switch self {
+        case .celsius:
+            return "Celsius"
+        case .fahrenheit:
+            return "Fahrenheit"
+        default:
+            return "Reserved for future use"
+        }
+    }
+}
+
+// See: https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.temperature_measurement.xml
+public class TemperatureMeasurementType {
+    private let characteristicData: [UInt8]
+
+    public var value: Float {
+        let bitValue = UInt32(characteristicData[1] << 24) +
+            UInt32(characteristicData[2] << 16) +
+            UInt32(characteristicData[3] << 8) +
+            UInt32(characteristicData[4])
+        return Float(bitPattern: bitValue)
+    }
+
+    public var unit: TemperatureMeasurementUnitEnum {
+        return characteristicData[0] & 0x01 == 0 ? .celsius : .fahrenheit
+    }
+
+    public var type: TemperatureMeasurementTypeEnum {
+        return TemperatureMeasurementTypeEnum.init(rawValue: UInt8(characteristicData[2])) ?? .unknown
```

Comment:
characteristicData[2] will never be for TemperatureMeasurementTypeEnum.
If no timestamp, it will be **5**
If timestamp, it will be **12**